app/backup.go

```diff
@@ -265,6 +265,10 @@ func doRestoreBackup(c *cli.Context) error {
 		return err
 	}
 	if err := task.RestoreBackup(backupURL); err != nil {
+		//Finish Restore as initiation failed
+		if err = cli.VolumeFinishRestore(""); err != nil {
```

Comment:
Do we need to call cli.VolumeFinishRestore() in this case? Normally it should be handled automatically by the task.RestoreBackup().

Reply:
Yes. This function is reverting the flag (c.IsRestoring in controller server) which was set by VolumePrepareRestore in line #263. I believe, this is required.